OrbitGl/Track.cpp

```diff
@@ -5,73 +5,128 @@
 #include "Track.h"
 
 #include "Capture.h"
+#include "CoreMath.h"
 #include "GlCanvas.h"
 #include "TimeGraphLayout.h"
 #include "absl/strings/str_format.h"
 
 //-----------------------------------------------------------------------------
-Track::Track(TimeGraph* time_graph) : time_graph_(time_graph) {
+Track::Track(TimeGraph* time_graph)
+    : time_graph_(time_graph),
+      collapse_toggle_(
+          true, [this](bool state) { OnCollapseToggle(state); }, time_graph) {
   m_MousePos[0] = m_MousePos[1] = Vec2(0, 0);
   m_Pos = Vec2(0, 0);
   m_Size = Vec2(0, 0);
   m_PickingOffset = Vec2(0, 0);
   m_Picked = false;
   m_Moving = false;
   m_Canvas = nullptr;
-  unsigned char alpha = 255;
-  unsigned char grey = 60;
-  m_Color = Color(grey, grey, grey, alpha);
 }
 
 //-----------------------------------------------------------------------------
-void Track::Draw(GlCanvas* a_Canvas, bool a_Picking) {
-  static volatile unsigned char alpha = 255;
-  static volatile unsigned char grey = 60;
-  auto col = Color(grey, grey, grey, alpha);
-  a_Picking ? PickingManager::SetPickingColor(
-                  a_Canvas->GetPickingManager().CreatePickableId(this))
-            : glColor4ubv(&m_Color[0]);
+std::vector<Vec2> GetRoundedCornerMask(float radius, uint32_t num_sides) {
+  std::vector<Vec2> points;
+  points.emplace_back(0.f, 0.f);
+  points.emplace_back(0.f, radius);
+
+  float increment_radians = 0.5f * M_PI / static_cast<float>(num_sides);
+  for (uint32_t i = 1; i < num_sides; ++i) {
+    float angle = M_PI + static_cast<float>(i) * increment_radians;
```

Comment:
This line creates a warning (which is treated as error) for me (double to float conversion).

Reply:
(And line 33 as well.)

Reply:
Added kPiFloat constant.